pkgs/top-level/all-packages.nix

```
@@ -14289,7 +14289,9 @@ in
 
   libmemcached = callPackage ../development/libraries/libmemcached { };
 
-  libmicrohttpd = callPackage ../development/libraries/libmicrohttpd { };
+  libmicrohttpd_0_9_70 = callPackage ../development/libraries/libmicrohttpd/0.9.70.nix { };
+  libmicrohttpd_0_9_71 = callPackage ../development/libraries/libmicrohttpd/0.9.71.nix { };
```

```suggestion
  libmicrohttpd_0_9_71 = callPackage ../development/libraries/libmicrohttpd/default.nix { };
```

And maybe we shouldn't use the generic file if they diverge in the future.

---

Personally a little unclear on when to use default.nix vs generic.nix in such scenarios - most libraries where the inputs and build process remains the same seem to use generic, but there are a bunch which use default as well (usually, those have varying build inputs too)

I could live with changing it back to default if people agree that's a better fit here

---

microhttpd doesn't see a whole lot of development - personally, I'd leave the generic derivation for now - the dependencies haven't really changed in a while.

If they do diverge, we could split it up, but I don't see a point in preemptively adding in extra things we may not need by splitting it up now